keymanagementscenarios.md

```diff
@@ -0,0 +1,105 @@
+- Enterprises can restrict users to add/remove root public keys for trusted third parties to their trust stores. The trust store will validate images pulled from any registry.
+- Trust store also includes subordinate keys
+- Air gapped environments
+
+Key rotation/revocation use cases:
+- Root Revocation (compromised root should not be needed in process to designate itself as revoked, otherwise attacker can use compromised root for a key rotation locking out publisher)
+    - Publisher deletes revoked root public key on registry (registry can stop sharing containers with revoked key).
+    - [TODO: Discuss if we want this] Registry stops vending containers signed with old root key (will prevent revoked artifacts from being used by developers not checking signature).
+    - Publisher removes revoked root public key from shared list (other users stop trusting artifacts with revoked key, provides defense in depth as they are not relying on registry).
+    - Publisher creates new root key following any scenario listed earlier in Signing use cases (starting from scratch as old key can no longer be trusted).
+    - Publisher uploads new root public key to registry (registry can verify user for container uploads). 
+    - Publisher updates shared root public key (other users can verify containers with new key before running).
+    - Publisher creates new delegate keys following any scenario listed earlier in Signing use cases.
+- Delegate Key Revocation (root can be used here for designating delegate keys as revoked as it has not been compromised)
+    - Publisher lists revoked delegate keys as untrusted. Revocation list is signed by root key (Design discussion on whether we use CRLs or some other approach to sharing information on revoked keys).
+    - Publisher shares revocation list (deployer needs to know which keys used prior for generating signatures are no longer trusted).
+    - Publisher creates new delegate keys from existing root following any scenario listed earlier in Signing use cases.
+- Root Rotation (periodic rotation of root will limit blast radius, this process should not revoke existing root as it has not been compromised)
+    - Publisher creates new root key following any scenario listed earlier in Signing use cases.
+    - Publisher uploads new root public key to registry (registry can verify user for container uploads). 
+    - Publisher updates shared root public key (other users can verify containers with new key before running).
+    - Publisher creates new delegate keys from new root following any scenario listed earlier in Signing use cases.
+- Delegate Key Rotation
+    - Publisher creates new delegate keys from existing root following any scenario listed earlier in Signing use cases.
+
+In addition we also need to consider the following for cryptographic security:
+- Supported key types
+- Supported signing algorithms
+
+Identify use cases to clarify:
```

Comment:
What about cryptographic algorithms? Do we wanted versioned protocols, or cryptographic agility?